xcode/Subconscious/Shared/Components/Detail/DetailMetaSheet.swift

```diff
@@ -29,13 +29,15 @@ struct DetailMetaSheet: View {
                         }
                     }
                     .font(.callout)
-                    AudienceMenuButtonView(
-                        audience: Binding(
-                            get: { state.audience },
-                            send: send,
-                            tag: DetailMetaSheetAction.requestUpdateAudience
+                    if AppDefaults.standard.isNoosphereEnabled {
```

Comment:
Even though there's no way for a user to toggle this setting while the sheet is up, I wanted to check if this re-renders when the `AppDefault` changes... and I was surprised to find it does re-render!?

Is this a byproduct of adding a publisher to these properties or do `AppDefaults` changes trigger magic renders?

Reply:
That's the publisher doing its thing. The subscription to `AppDefaults.standard.$isNoosphereEnabled` means we get an event sent to the store whenever the value changes. This then replays the value change to the store and updates the store's copy of the property, triggering re-render.